### PR TITLE
Basic architecture interfaces

### PR DESCRIPTION
**Pull request by EdenComp**
This PR contains basic interfaces for the architecture.

Any modification of them will require a new PR.

### CODE REVIEW

File: include/common/Elements.hpp [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+namespace common {
+    struct Position {
+        std::size_t x;
+        std::size_t y;
+    };
+
+    struct Sprite {
+        Position position;
+        std::string filepath;
+        char character;
```
Comment (RezaRahemtola) [inline include/common/Elements.hpp:22]: remove this line

File: include/common/Input.hpp [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+** EPITECH PROJECT, 2023
+** Arcade-Common
+** File description:
+** Input.hpp
+*/
+
+#ifndef INPUT_HPP_
+#define INPUT_HPP_
+
+namespace common {
```
Comment (RezaRahemtola) [inline include/common/Input.hpp:11]: same for namespace

File: include/common/Input.hpp [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    enum class Input {
+        UP,
+        DOWN,
+        RIGHT,
+        LEFT,
+        ACTION_1,
```
Comment (Steci) [inline include/common/Input.hpp:17]: No Fire ?

**Review by Steci — APPROVED** [review]

**Review by Tomi-Tom — APPROVED** [review]
Noice Work Matesse ☕

**Review by RezaRahemtola — DISMISSED** [review]

File: include/common/Elements.hpp [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+** File description:
+** Elements.hpp
+*/
+
+#ifndef ELEMENTS_HPP_
+#define ELEMENTS_HPP_
+
+#include <string>
+
+namespace common {
```
[the remark this answers is not in the document]
Comment (EdenComp) [reply include/common/Elements.hpp]: Done

File: include/common/display/IDisplayModule.hpp [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+** File description:
+** IDisplayModule.hpp
+*/
+
+#ifndef IDISPLAYMODULE_HPP_
+#define IDISPLAYMODULE_HPP_
+
+#include "common/Elements.hpp"
+
+namespace common {
```
Comment (Croos3r) [inline include/common/display/IDisplayModule.hpp]: arcade::common ?

**Review by Croos3r — CHANGES_REQUESTED** [review]

File: include/common/Elements.hpp (hunk 1 above)
Comment (EdenComp) [reply include/common/Elements.hpp]: Done

**Review by Steci — APPROVED** [review]

**Review by Tomi-Tom — APPROVED** [review]
Oui oui oui

**Comment by Tomi-Tom** [comment]
STOP LES MAILS ⚠️

File: include/common/Input.hpp [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+        ACTION_2,
+        SPECIAL_1,
+        SPECIAL_2,
```
[the remark this answers is not in the document]
Comment (EdenComp) [reply include/common/Input.hpp:23]: Done

File: include/common/Input.hpp (hunk 3 above)
Comment (RezaRahemtola) [reply include/common/Input.hpp:20]: action is fine

**Review by Steci — APPROVED** [review]

**Review by Tomi-Tom — APPROVED** [review]
Gutt Gutt

**Review by Chasfory — APPROVED** [review]

**Pull request merged**